virtual_rainforest/core/config.py

```diff
@@ -315,11 +386,42 @@ def construct_combined_schema(modules: list[str]) -> dict:
     return comb_schema
 
 
+def validate_with_defaults(config_dict: dict, comb_schema: dict) -> dict:
+    """Validate the configuration settings against the combined schema.
+
+    This function also adds default values into the configuration dictionary where it is
+    appropriate.
+
+     Args:
+        config_dict: The complete configuration settings for the particular model
+            instance
+        comb_schema: Combined schema for all modules that are being configured
+
+    Raises:
+        RuntimeError: If the configuration files fail to validate against the JSON
+            schema
+    """
+
+    # Make a new validator that allows the addition of defaults
+    ValidatorWithDefaults = extend_with_default(Draft202012Validator)
```

Comment:
Is there any reason you can't extend `Draft202012Validator` by inheritance (i.e. `class ValidatorWithDefaults(Draft202012Validator)`?

Reply:
Going one step further, I'd be inclined to make this a member method of your `ValidatorWithDefaults` class.

Minor nit: I found the name a bit confusing. How about `validate_and_add_defaults()`? It's a bit cumbersome but conveys that `config_dict` is being modified

Reply:
Having a bit of difficultly with this one. Basically `jsonschema` already creates a class structure for validators, with the function `validators.extend` being used to create new class instances. Not really sure how to define my own `ValidatorWithDefaults` and add in `validate_and_add_defaults()` as a member function without overwriting the class structure defined by the `jsonschema` package? As an alternative I was thinking of keeping `validate_and_add_defaults()` where it is, but creating the Validator once at global scope. This stops `ValidatorWithDefaults` from having to be created multiple times

Reply:
Okey doke. I didn't realise that's how ``jsonschema`` does things, but if that's the case then it's ok by me!